### PR TITLE
[feature] Optimize computeErasParse and computeMonthsParse

### PR DESCRIPTION
**Pull request by Alanscut**
## summary
 Optimize `computeErasParse` and `computeMonthsParse` , reduce unnecessary regular escaping  and loop traversal.

### CODE REVIEW

File: src/lib/units/month.js [hunk 1]
```diff
@@ -314,27 +314,24 @@ function computeMonthsParse() {
...
+        longPieces.push(longP);
+        mixedPieces.push(longP);
+        mixedPieces.push(shortP);
     }
     // Sorting makes sure if one month (or abbr) is a prefix of another it
     // will match the longer piece.
     shortPieces.sort(cmpLenRev);
     longPieces.sort(cmpLenRev);
     mixedPieces.sort(cmpLenRev);
-    for (i = 0; i < 12; i++) {
```
Comment (marwahaha) [inline src/lib/units/month.js:331]: This could change functionality, correct?

It's first a sort, then escape. Your change escapes, then sorts.

Comment (Alanscut) [reply src/lib/units/month.js:331]: `regexEscape` function may change the length of shortP, longP, and mixedP, leading a wrong sorting, so I changed the order of escapes and sorts

Comment (marwahaha) [reply src/lib/units/month.js:331]: Do you ever want to sort before you escape the regex? 
Does it make sense to test this behavior?

**Comment by ichernev** [comment]
Merged in a3babc50afba073cb377ef7ce4b77803e928078c

**Comment by ichernev** [comment]
I tweaked a bit the eras code, was double-escaping the regex for the mixed case.

**Pull request closed without merging**